### ISSUE TITLE
Real Life Example

### ISSUE THREAD

**Issue by defuse** [defuse/php-encryption#16]
Make a real life example, showing, most importantly, how error handling should be done.

**Comment by paragonie-scott** [comment]
You can count [password_lock](https://github.com/paragonie/password_lock) as a real-life example.

**Comment by defuse** [comment]
I like password_lock as a real-life example, except that it doesn't demonstrate error handling. Is anyone using this in an actual project (i.e. not as a dependency in another library)?

**Comment by hursey013** [comment]
I am new to web development, so as part of trying to become more proficient at PHP I built a simple frontend for php-encryption called [smokescrn](https://github.com/hursey013/smokescrn/) that provides self destructing encrypted messages.  Here is the relevant code for [encrypting](https://github.com/hursey013/smokescrn/blob/master/encrypt.php#L99-L138) and [decrypting](https://github.com/hursey013/smokescrn/blob/master/decrypt.php#L78-L105).  I know it's pretty basic, but if there aren't any glaring security holes in my implementation, I'd love to continue building upon it to be used as a real world example - [smokescrn.com](https://smokescrn.com).  Let me know what you think, and thanks for all of your work developing this library!

**Comment by pnowosie** [comment]
@hursey013 I like your website it looks really pretty. I'd recommend using Crypto::binToHex() instead of base64_encode() when you store a message. It might make sense for demonstration purposes.

What "SCRN" comes from?

**Comment by paragonie-scott** [comment]
(I'm assuming it's an abbreviation for screen)

**Comment by hursey013** [comment]
@pnowosie thanks for the kind words.  Yes, SCRN is just an abbreviation for screen - it's getting harder and harder to find a good domain name, so I had to get a little creative with the spelling!

I took your suggestion and [implemented binToHex and hexToBin](https://github.com/hursey013/smokescrn/commit/9656a3eba549ae5ac9795bd6b10df8a96be565d7).  I think I made it more difficult than it needed to be... I didn't realize that those functions weren't included in ver. 1.2.1 so I kept getting errors.  Once I switched composer to use the dev-master branch that did the trick!  Thanks again for your input and suggestion.
